Source/WebCore/dom/MouseEvent.cpp

```diff
@@ -80,36 +80,36 @@ MouseEvent::MouseEvent() = default;
     MonotonicTime timestamp, RefPtr<WindowProxy>&& view, int detail,
-    const IntPoint& screenLocation, const IntPoint& windowLocation, double movementX, double movementY, OptionSet<Modifier> modifiers, short button, unsigned short buttons,
+    const IntPoint& screenLocation, const IntPoint& windowLocation, double movementX, double movementY, OptionSet<Modifier> modifiers, MouseButton button, unsigned short buttons,
     EventTarget* relatedTarget, double force, SyntheticClickType syntheticClickType, IsSimulated isSimulated, IsTrusted isTrusted)
     : MouseRelatedEvent(eventType, canBubble, isCancelable, isComposed, timestamp, WTFMove(view), detail, screenLocation, windowLocation, movementX, movementY, modifiers, isSimulated, isTrusted)
-    , m_button(button == -2 ? 0 : button)
+    , m_button(button == MouseButton::NoButton ? MouseButton::LeftButton : button)
     , m_buttons(buttons)
-    , m_syntheticClickType(button == -2 ? SyntheticClickType::NoTap : syntheticClickType)
-    , m_buttonDown(button != -2)
+    , m_syntheticClickType(button == MouseButton::NoButton ? SyntheticClickType::NoTap : syntheticClickType)
+    , m_buttonDown(button != MouseButton::NoButton)
     , m_relatedTarget(relatedTarget)
     , m_force(force)
 {
 }
 
 MouseEvent::MouseEvent(const AtomString& eventType, CanBubble canBubble, IsCancelable isCancelable, IsComposed isComposed,
     RefPtr<WindowProxy>&& view, int detail, const IntPoint& screenLocation, const IntPoint& clientLocation, double movementX, double movementY,
-    OptionSet<Modifier> modifiers, short button, unsigned short buttons, SyntheticClickType syntheticClickType, EventTarget* relatedTarget)
+    OptionSet<Modifier> modifiers, MouseButton button, unsigned short buttons, SyntheticClickType syntheticClickType, EventTarget* relatedTarget)
     : MouseRelatedEvent(eventType, canBubble, isCancelable, isComposed, MonotonicTime::now(), WTFMove(view), detail, screenLocation, { }, movementX, movementY, modifiers, IsSimulated::No)
-    , m_button(button == -2 ? 0 : button)
+    , m_button(button == MouseButton::NoButton ? MouseButton::LeftButton : button)
     , m_buttons(buttons)
-    , m_syntheticClickType(button == -2 ? SyntheticClickType::NoTap : syntheticClickType)
-    , m_buttonDown(button != -2)
+    , m_syntheticClickType(button == MouseButton::NoButton ? SyntheticClickType::NoTap : syntheticClickType)
+    , m_buttonDown(button != MouseButton::NoButton)
     , m_relatedTarget(relatedTarget)
 {
     initCoordinates(clientLocation);
 }
 
 MouseEvent::MouseEvent(const AtomString& eventType, const MouseEventInit& initializer)
     : MouseRelatedEvent(eventType, initializer)
-    , m_button(initializer.button == -2 ? 0 : initializer.button)
+    , m_button(static_cast<MouseButton>(initializer.button) == MouseButton::NoButton ? MouseButton::LeftButton : static_cast<MouseButton>(initializer.button))
```

Comment:
Hm... I think this is probably fine, but it might be worth double checking with some of our security folks (i.e. whether this constitutes UB).

Reply:
(It seems like we should _probably_ avoid this after all)

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/dac58bfd6a4a82417491d6272fb7f66ba081b565